test/addressService.test.ts

```diff
@@ -15,3 +44,201 @@ test("count: null request - expect error returned", async () => {
   expect(error).toBeDefined();
   expect(error).toEqual(NULL_ADDRESS_REQUEST_ERROR);
 });
+
+test("count: bad zipcode - expect count of 0", async () => {
+  await addressService
+    .count({ body: { zipcode: "12" } })
+    .then((response) => {
+      expect(response).toBeDefined();
+      expect(response.count).toEqual(0);
+    })
+    .catch((err) => {
+      fail("addressService.count - Unexpected Error: " + err);
+    });
+}, 90000);
+
+test("distance: null request - expect error returned", async () => {
+  let error: any;
+  await addressService
+    .distance(null)
+    .then(() => {
+      fail("addressService.distance: Unexpected Success");
+    })
+    .catch((err: Error) => {
+      error = err;
+    });
+
+  expect(error).toBeDefined();
+  expect(error.message).toEqual(NULL_ADDRESS_REQUEST_ERROR);
+}, 60000);
+
+test("distance: bad request - expect error returned", async () => {
+  let error: any;
+  await addressService
+    .distance({
+      body: {
+        addresses: [
+          {
+            number: "4255",
+            // Should be "E RIVER RD", this will send error
+            street: "E RIVER ROAD",
+            city: "WEST HENRIETTA",
+            state: "NY",
+            zipcode: "14586",
+          },
+          {
+            number: "4255",
+            street: "E RIVER RD",
+            street2: "18C",
+            city: "WEST HENRIETTA",
+            state: "NY",
+            zipcode: "14586",
+          },
+        ],
+      },
+    })
+    .then(() => {
+      fail("addressService.distance: Unexpected Success");
+    })
+    .catch((err: Error) => {
+      error = err;
+    });
+
+  expect(error).toBeDefined();
+  expect(error.message).toEqual(ADDRESS_NOT_FOUND_ERROR);
+});
+
+test("distance: good request - expect distance returned", async () => {
+  const expectedKilometers = 0.07437772516419937;
+  const expectedMiles = 0.0462161755947715;
+
+  let actualKilometers;
+  let actualMiles;
+
+  await addressService
+    .distance({
+      body: {
+        addresses: [
+          {
+            number: "4255",
+            street: "E RIVER RD",
+            street2: "15C",
+            city: "WEST HENRIETTA",
+            state: "NY",
+            zipcode: "14586",
+          },
+          {
+            number: "4255",
+            street: "E RIVER RD",
+            street2: "18C",
+            city: "WEST HENRIETTA",
+            state: "NY",
+            zipcode: "14586",
+          },
+        ],
+      },
+    })
+    .then((distances) => {
+      actualKilometers = distances[0];
+      actualMiles = distances[1];
+    })
+    .catch((err) => {
+      fail("addressService.distance: Unexpected Error - " + err.message);
+    });
+
+  expect(actualKilometers).toBeDefined();
+  expect(actualKilometers).toEqual(expectedKilometers);
+
+  expect(actualMiles).toBeDefined();
+  expect(actualMiles).toEqual(expectedMiles);
```

Comment:
Same as above. If someone comes by and removes these awaits thinking they're speeding up tests, the assertions will fire before the promise finishes executing.